indra/llmessage/lltemplatemessagereader.cpp

```diff
@@ -286,9 +286,9 @@ void LLTemplateMessageReader::getU8(const char *block, const char *var,
 void LLTemplateMessageReader::getBOOL(const char *block, const char *var, 
 										  bool &b, S32 blocknum )
 {
-	U8 value;
-	getData(block, var, &value, sizeof(U8), blocknum);
-	b = (BOOL) value;
+	bool value;
+	getData(block, var, &value, sizeof(bool), blocknum);
```

Comment:
Does this actually work? I'm concerned the simulator is most likely sending U8 for bool and that it might break if you now just read less data than expected. I would probably keep reading and U8 and then use a static_cast to convert it to bool.

Reply:
Reverted so we cast again.